### ISSUE TITLE
How do I access the properties of parent component in the custom component's onComponentInitFunction

### ISSUE THREAD

**Issue by EtiAggarwal** [akveo/ng2-smart-table#401]
I need to emit an event when an event occurs in the custom component.
  onComponentInitFunction (instance: any) {
                    instance.cellSelected.subscribe((cust: ICustomer) => {
                       // this.customerSelected.emit(cust); -- need a way to raise an event here
                    });
                },

**Comment by EtiAggarwal** [comment]
Got it to work.

**Comment by Crizstian** [comment]
Could you provide an example in how do you achieve it ?

**Comment by EtiAggarwal** [comment]
I used lambda expression syntax for onComponentInitFunction that gave access to "this"
onComponentInitFunction: (instance: any) => {
                        this.selectedCustomer = cust;
                        this.customerSelected.emit(cust);
                    });
                }

**Comment by wooltar** [comment]
```
 button: {
        title: 'Actions',
        filter: false,
        type: 'custom',
        renderComponent: ActionButtonComponent,
        onComponentInitFunction(instance) {
          instance.save.subscribe(row => {
             console.log(row);
          });
        }
}
```
 ```

@Component({
  selector: 'app-action-button',
  templateUrl: './action-button.component.html',
  styleUrls: ['./action-button.component.scss']
})
export class ActionButtonComponent implements ViewCell, OnInit {
  renderValue: string;
  @Input() value: string | number;
  @Input() rowData: any;

  @Output() save: EventEmitter<any> = new EventEmitter();

  ngOnInit() {
    this.renderValue = this.value.toString().toUpperCase();
  }

  onClick(row) {
    this.save.emit(row);
  }
}
```

**Comment by FinalButterfly** [comment]
Guys, thanks for this. It's 2019, but hey, version 1.4 of ng2-smart-table still poses the same problem. Spent hours trying to do it.

**Comment by ccruza** [comment]
I achieve to set a variable value of the renderComponent.
So this is my column:

`columns: {
                Receipt_Reference: {
                    title: 'Receipt Ref.',
                    filter: true,
                    editable: false,
                    addable: false,
                    type: 'custom',
                    renderComponent: TooltipComponent,
                    onComponentInitFunction(instance) {
                        instance.TooltipText = 'This is the tooltip text I want to pass';
                    }
                },`


then in my TooltipComponent:

`import { Component, OnInit, Input, EventEmitter, Output, NgModule } from '@angular/core';

@Component({
    selector: 'tooltip-view',
    // template: `<div class="nodelist" ngbTooltip="You see, I show up on click!">{{rowData.Receipt_Reference}}</div>`,
    template: `<div class="nodelist" ngbTooltip={{TooltipText}}>{{rowData.Receipt_Reference}}</div>`,
})

@NgModule()
export class TooltipComponent{
    renderValue: string;
    TooltipText : string = 'Esto es un tooltip';

    @Input() value: string | number;
    @Input() rowData: any;

}
`

And this is the result:

![image](https://user-images.githubusercontent.com/41245212/57898350-498f7080-781e-11e9-9d65-684fe0b295f8.png)

Hope this Will help you

**Comment by FinalButterfly** [comment]
I solved the problem a few weeks ago. However, while it works the same, I took what I thought was a cleaner code approach. My table settings are defined in a separate file so they don't irritate my eye being in the component. The column in question is defined as empty ('manage' in my case)

![image](https://user-images.githubusercontent.com/31434462/57911223-97e55380-78a0-11e9-9c1d-57aab113b71c.png)

Then, in the controller ngOnInit(), I define the functionality for the column with all the bells and whistles.

![image](https://user-images.githubusercontent.com/31434462/57911272-c400d480-78a0-11e9-9d99-738f967ecf92.png)
